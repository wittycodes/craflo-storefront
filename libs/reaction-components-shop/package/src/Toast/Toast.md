### Overview

Toasts are used to give action-based feedback messages and convey critical or informational account-related messages. Use Toasts when a user needs more detailed information for an action.

The Toast component inherits from the Material-UI [Snackbar component](https://material-ui.com/components/snackbars/). Refer to the Material-UI [Snackbar docs](https://material-ui.com/api/snackbar/) for more information.

### Usage

Toasts are most often used when the user has taken an action. Messages appear in context and communicate when that action is successful, unsuccessful, or that it otherwise needs attention and further context.

Language should be polite, clear and concise. Optionally, a title can be added to a Toast to give clarity, or when there are 2 or more lines of information to display.

Toasts should guide the user into taking corrective action if necessary.

Users should be able to dismiss Toasts when appropriate. Information and success alerts can close automatically after 10 seconds. Error alerts should be persistent, and close only when action is resolved.

#### Types

##### Default: Information

- Used when there is information or tips that users can benefit from
- Can close automatically after 10 seconds

```jsx
import Button from "../Button";
import IconButton from "@material-ui/core/IconButton";

function OpenToast(props) {
  const [open, setOpen] = React.useState(false);

  const handleClose = (event, reason) => {
    if (reason === 'clickaway') {
      return;
    }

    setOpen(false);
  };

  return (
    <div>
      <Button variant="contained" color="primary" onClick={() => setOpen(true)}>Open information toast</Button>
      <Button variant="outlined" color="primary" onClick={() => setOpen(false)}>Close information toast</Button>
      <Toast
        open={open}
        onClose={handleClose}
        message={props.message}
        variant={props.variant}
        title={props.title}
      />
    </div>
  );
}

<OpenToast message={<span>Information toast</span>} />
```
##### Success

- Used when an action has been completed successfully
- Can close automatically after 10 seconds

```jsx
import Button from "../Button";
import IconButton from "@material-ui/core/IconButton";

function OpenToast(props) {
  const [open, setOpen] = React.useState(false);

  const handleClose = (event, reason) => {
    if (reason === 'clickaway') {
      return;
    }

    setOpen(false);
  };

  return (
    <div>
      <Button variant="contained" color="primary" onClick={() => setOpen(true)}>Open success toast</Button>
      <Button variant="outlined" color="primary" onClick={() => setOpen(false)}>Close success toast</Button>
      <Toast
        open={open}
        onClose={handleClose}
        message={props.message}
        variant={props.variant}
        title={props.title}
      />
    </div>
  );
}

<OpenToast message={<span>Success toast</span>} variant="success" />
```

##### Warning

- Used when an action or item needs attention
- Should not close automatically, unless the action has been resolved

```jsx
import Button from "../Button";
import IconButton from "@material-ui/core/IconButton";

function OpenToast(props) {
  const [open, setOpen] = React.useState(false);

  const handleClose = (event, reason) => {
    if (reason === 'clickaway') {
      return;
    }

    setOpen(false);
  };

  return (
    <div>
      <Button variant="contained" color="primary" onClick={() => setOpen(true)}>Open warning toast</Button>
      <Button variant="outlined" color="primary" onClick={() => setOpen(false)}>Close warning toast</Button>
      <Toast
        open={open}
        onClose={handleClose}
        message={props.message}
        variant={props.variant}
        title={props.title}
      />
    </div>
  );
}

<OpenToast message={<span>Warning toast</span>} variant="warning" />
```
##### Error

- Used when the system has failed to complete an action, or the user has made an error
- Should not close automatically, unless the action has been resolved

```jsx
import Button from "../Button";
import IconButton from "@material-ui/core/IconButton";

function OpenToast(props) {
  const [open, setOpen] = React.useState(false);

  const handleClose = (event, reason) => {
    if (reason === 'clickaway') {
      return;
    }

    setOpen(false);
  };

  return (
    <div>
      <Button variant="contained" color="error" onClick={() => setOpen(true)}>Open error toast</Button>
      <Button variant="outlined" color="error" onClick={() => setOpen(false)}>Close error toast</Button>
      <Toast
        open={open}
        onClose={handleClose}
        message={props.message}
        variant={props.variant}
        title={props.title}
      />
    </div>
  );
}

<OpenToast message={<span>Error toast</span>} variant="error" />
```

#### Toasts with Titles

- All toasts can also have a `Title`

```jsx
import Button from "../Button";
import IconButton from "@material-ui/core/IconButton";

function OpenToast(props) {
  const [open, setOpen] = React.useState(false);

  const handleClose = (event, reason) => {
    if (reason === 'clickaway') {
      return;
    }

    setOpen(false);
  };

  return (
    <div>
      <Button variant="contained" color="primary" onClick={() => setOpen(true)}>Open title toast</Button>
      <Button variant="outlined" color="primary" onClick={() => setOpen(false)}>Close title toast</Button>
      <Toast
        open={open}
        onClose={handleClose}
        message={props.message}
        variant={props.variant}
        title={props.title}
      />
    </div>
  );
}

<OpenToast message={<span>Information toast with title</span>} title="Title" />
```