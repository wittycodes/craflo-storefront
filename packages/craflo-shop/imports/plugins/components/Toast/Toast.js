"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _core = require("@material-ui/core");

var _helpers = require("./helpers");

/**
 * @name Toast
 * @param {Object} props Component props
 * @returns {React.Component} returns a React component
 */
var Toast = _react.default.forwardRef(function Toast(props, ref) {
  var message = props.message,
      variant = props.variant,
      title = props.title,
      onClose = props.onClose,
      otherProps = (0, _objectWithoutProperties2.default)(props, ["message", "variant", "title", "onClose"]);
  return /*#__PURE__*/_react.default.createElement(_core.Snackbar, (0, _extends2.default)({
    ref: ref
  }, otherProps), /*#__PURE__*/_react.default.createElement(_helpers.ToastWrapper, {
    props: otherProps,
    variant: variant,
    title: title,
    message: message,
    onClose: onClose
  }));
});

Toast.propTypes = {
  /**
   * Message: Node, <span>Message goes here</span>
   */
  message: _propTypes.default.node,

  /**
   * onClose: Callback fired when the component requests to be closed and when the X icon button is clicked
   */
  onClose: _propTypes.default.func.isRequired,

  /**
   * Title: Optional string, displayed in bold
   */
  title: _propTypes.default.string,

  /**
   * Variant: Info, Success, Warning, Error
   */
  variant: _propTypes.default.oneOf(["error", "info", "success", "warning"]).isRequired
};
Toast.defaultProps = {
  variant: "info"
};
var _default = Toast;
exports.default = _default;