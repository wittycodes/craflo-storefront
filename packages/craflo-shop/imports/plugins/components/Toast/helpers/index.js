"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ToastWrapper", {
  enumerable: true,
  get: function get() {
    return _ToastWrapper.default;
  }
});

var _ToastWrapper = _interopRequireDefault(require("./ToastWrapper"));