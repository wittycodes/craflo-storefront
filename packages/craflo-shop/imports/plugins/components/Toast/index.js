"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "default", {
  enumerable: true,
  get: function get() {
    return _Toast.default;
  }
});

var _Toast = _interopRequireDefault(require("./Toast"));