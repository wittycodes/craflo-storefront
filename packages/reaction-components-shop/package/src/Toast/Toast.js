import React from "react";
import PropTypes from "prop-types";
import { Snackbar } from "@material-ui/core";
import { ToastWrapper } from "./helpers";

/**
 * @name Toast
 * @param {Object} props Component props
 * @returns {React.Component} returns a React component
 */
const Toast = React.forwardRef(function Toast(props, ref) {
  const { message, variant, title, onClose, ...otherProps } = props;

  return (
    <Snackbar
      ref={ref}
      {...otherProps}
    >
      <ToastWrapper
        props={otherProps}
        variant={variant}
        title={title}
        message={message}
        onClose={onClose}
      />
    </Snackbar>
  );
});


Toast.propTypes = {
  /**
   * Message: Node, <span>Message goes here</span>
   */
  message: PropTypes.node,
  /**
   * onClose: Callback fired when the component requests to be closed and when the X icon button is clicked
   */
  onClose: PropTypes.func.isRequired,
  /**
   * Title: Optional string, displayed in bold
   */
  title: PropTypes.string,
  /**
   * Variant: Info, Success, Warning, Error
   */
  variant: PropTypes.oneOf(["error", "info", "success", "warning"]).isRequired
};

Toast.defaultProps = {
  variant: "info"
};

export default Toast;
