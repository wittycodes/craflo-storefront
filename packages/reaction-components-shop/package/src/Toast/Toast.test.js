import React from "react";
import { render } from "../tests/index.js";
import Toast from "./Toast";

test("basic snapshot - only default props", () => {
  const { asFragment } = render(<Toast message="Test message" open/>);
  expect(asFragment()).toMatchSnapshot();
});

test("basic snapshot - information variant", () => {
  const { asFragment } = render(<Toast message="Test message" variant="info" open/>);
  expect(asFragment()).toMatchSnapshot();
});

test("basic snapshot - success variant", () => {
  const { asFragment } = render(<Toast message="Test message" variant="success" open/>);
  expect(asFragment()).toMatchSnapshot();
});

test("basic snapshot - error variant", () => {
  const { asFragment } = render(<Toast message="Test message" variant="error" open/>);
  expect(asFragment()).toMatchSnapshot();
});

test("basic snapshot - warning variant", () => {
  const { asFragment } = render(<Toast message="Test message" variant="warning" open/>);
  expect(asFragment()).toMatchSnapshot();
});
