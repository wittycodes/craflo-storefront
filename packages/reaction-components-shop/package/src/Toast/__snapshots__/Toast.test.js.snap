// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot - error variant 1`] = `
<DocumentFragment>
  <div
    class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomCenter"
  >
    <div
      aria-describedby="message-id"
      class="MuiPaper-root makeStyles-error-235 MuiPaper-elevation6"
      direction="up"
      props="[object Object]"
      role="alertdialog"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="makeStyles-messageWrapper-231"
      >
        Test message
      </div>
      <button
        aria-label="close"
        class="MuiButtonBase-root MuiIconButton-root makeStyles-close-230"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`basic snapshot - information variant 1`] = `
<DocumentFragment>
  <div
    class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomCenter"
  >
    <div
      aria-describedby="message-id"
      class="MuiPaper-root makeStyles-info-88 MuiPaper-elevation6"
      direction="up"
      props="[object Object]"
      role="alertdialog"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="makeStyles-messageWrapper-83"
      >
        Test message
      </div>
      <button
        aria-label="close"
        class="MuiButtonBase-root MuiIconButton-root makeStyles-close-82"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`basic snapshot - only default props 1`] = `
<DocumentFragment>
  <div
    class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomCenter"
  >
    <div
      aria-describedby="message-id"
      class="MuiPaper-root makeStyles-info-14 MuiPaper-elevation6"
      direction="up"
      props="[object Object]"
      role="alertdialog"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="makeStyles-messageWrapper-9"
      >
        Test message
      </div>
      <button
        aria-label="close"
        class="MuiButtonBase-root MuiIconButton-root makeStyles-close-8"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`basic snapshot - success variant 1`] = `
<DocumentFragment>
  <div
    class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomCenter"
  >
    <div
      aria-describedby="message-id"
      class="MuiPaper-root makeStyles-success-160 MuiPaper-elevation6"
      direction="up"
      props="[object Object]"
      role="alertdialog"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="makeStyles-messageWrapper-157"
      >
        Test message
      </div>
      <button
        aria-label="close"
        class="MuiButtonBase-root MuiIconButton-root makeStyles-close-156"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`basic snapshot - warning variant 1`] = `
<DocumentFragment>
  <div
    class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomCenter"
  >
    <div
      aria-describedby="message-id"
      class="MuiPaper-root makeStyles-warning-311 MuiPaper-elevation6"
      direction="up"
      props="[object Object]"
      role="alertdialog"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="makeStyles-messageWrapper-305"
      >
        Test message
      </div>
      <button
        aria-label="close"
        class="MuiButtonBase-root MuiIconButton-root makeStyles-close-304"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
